### PR TITLE
allow disabling of Google Container Registry credential checks

### PR DESCRIPTION
**Pull request by mattnworb**
This is a continuation of #35; I added a commit addressing my own feedback to be able to merge and release this quicker.

### CODE REVIEW

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java [hunk 1]
```diff
@@ -401,13 +409,17 @@ private RegistryAuthSupplier createRegistryAuthSupplier() {
...
     suppliers.add(new ConfigFileRegistryAuthSupplier());
 
-    try {
-      final RegistryAuthSupplier googleSupplier = googleContainerRegistryAuthSupplier();
-      if (googleSupplier != null) {
-        suppliers.add(0, googleSupplier);
-      }
-    } catch (IOException ex) {
-      getLog().info("ignoring exception while loading Google credentials", ex);
+    if (googleContainerRegistryEnabled){
```
Comment (dflemstr) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: Missing space in front of `{` ;)

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java [hunk 2]
```diff
@@ -401,13 +409,17 @@ private RegistryAuthSupplier createRegistryAuthSupplier() {
...
+       try {
+         final RegistryAuthSupplier googleSupplier = googleContainerRegistryAuthSupplier();
+         if (googleSupplier != null) {
+           suppliers.add(0, googleSupplier);
+         }
+       } catch (IOException ex) {
+         getLog().info("ignoring exception while loading Google credentials", ex);
+       }
+    } else {
+      getLog().info("Google Container Registry support is disabled");
```
Comment (dflemstr) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: `.debug()`?

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java [hunk 3]
```diff
@@ -192,6 +192,14 @@ public String getFileName() {
   private JarArchiver jarArchiver;
 
   /**
+   * Allows disabling of Google Container Registry authentication support. The support is enabled by
+   * default, and should be a no-op (and fail fast) in most non-GCR environments, but this behavior
+   * can be explicitly disabled with this property if needed.
+   */
+  @Parameter(defaultValue = "true", property = "dockerfile.googleContainerRegistryEnabled")
+  private boolean googleContainerRegistryEnabled = true;
```
Comment (dflemstr) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: No need to have both `defaultValue = ...` and assign a default value

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java [hunk 4]
```diff
@@ -401,13 +409,17 @@ private RegistryAuthSupplier createRegistryAuthSupplier() {
...
-    } catch (IOException ex) {
-      getLog().info("ignoring exception while loading Google credentials", ex);
+    if (googleContainerRegistryEnabled){
+       try {
+         final RegistryAuthSupplier googleSupplier = googleContainerRegistryAuthSupplier();
+         if (googleSupplier != null) {
+           suppliers.add(0, googleSupplier);
+         }
+       } catch (IOException ex) {
+         getLog().info("ignoring exception while loading Google credentials", ex);
```
Comment (dflemstr) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: Uppercase `i` in `ignoring`

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java (hunk 2 above)
Comment (mattnworb) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: there is an info line already in googleContainerRegistryAuthSupplier when credentials are successfully loaded, so i wanted to have some parity

**Comment by mattnworb** [comment]
I released version 1.3.3 of the plugin a few minutes ago with this change in it.

**Pull request merged**